File: src/Domain/Models/General/RequestsResults/GenerateStub.cs
using Domain.Enums;

namespace Domain.Models.General.RequestsResults;

public class GenerateStubRequest
{
    public string Language { get; set; }
    public string Input { get; set; }
}

public class GenerateStubResult
{
    public string Stub { get; set; }

    public GenerateStubError Error { get; set; }
}

File: src/react-frontend/src/app/api/stubGenerator/types/generateStub.ts
export type GenerateStubError = {
  line: number;
  validationCode: string;
  culpritName: string;
};

export type GenerateStubRequest = {
  language: string;
  input?: string;
};

export type GenerateStubResult = {
  stub: string;
  error?: GenerateStubError;
};
